src/js/core.js

```diff
@@ -370,81 +373,41 @@ function MediumEditor(elements, options) {
         }
     }
 
-    function initPlaceholder(options) {
-        return new MediumEditor.extensions.placeholder(
-            Util.extend({}, options)
-        );
-    }
-
-    function initAnchorPreview(options) {
-        return new MediumEditor.extensions.anchorPreview(
-            Util.extend({}, options)
-        );
-    }
+    function initExtensions() {
 
-    function initAnchorForm(options) {
-        return new MediumEditor.extensions.anchor(
-            Util.extend({}, options)
-        );
-    }
+        this.extensions = [];
 
-    function initPasteHandler(options) {
-        return new MediumEditor.extensions.paste(
-            Util.extend({}, options)
-        );
-    }
-
-    function initCommands() {
-        var buttons = this.options.buttons,
-            extensions = this.options.extensions,
-            ext,
-            name;
-        this.commands = [];
-
-        buttons.forEach(function (buttonName) {
-            if (extensions[buttonName]) {
-                ext = initExtension(extensions[buttonName], buttonName, this);
-                this.commands.push(ext);
-            } else if (buttonName === 'anchor') {
-                ext = initExtension(initAnchorForm.call(this, this.options.anchor), 'anchor', this);
-                this.commands.push(ext);
-            } else if (buttonName === 'fontsize') {
-                ext = initExtension(new MediumEditor.extensions.fontSize(), buttonName, this);
-                this.commands.push(ext);
-            } else if (ButtonsData.hasOwnProperty(buttonName)) {
-                ext = initExtension(new MediumEditor.extensions.button(ButtonsData[buttonName]), buttonName, this);
-                this.commands.push(ext);
+        // Passed in extensions
+        Object.keys(this.options.extensions).forEach(function (name) {
+            // Always save the toolbar extension for last
+            if (name !== 'toolbar' && this.options.extensions[name]) {
+                this.extensions.push(initExtension(this.options.extensions[name], name, this));
             }
         }, this);
 
-        for (name in extensions) {
-            if (extensions.hasOwnProperty(name) && buttons.indexOf(name) === -1) {
-                ext = initExtension(extensions[name], name, this);
-                this.commands.push(ext);
+        // Built-in extensions
+        var builtIns = {
+            paste: true,
```

Comment:
We could, but the tricky thing with paste is that we have 2 methods (`cleanPaste()` + `pasteHTML()`) which require the paste handler to exist.  Without breaking the pasteHandler into 2 extensions (one for inserting HTML, 1 for handling pastes) I wasn't sure of another way to handle this other than always enforcing that a pasteHandler be present...